erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -413,8 +413,12 @@ def set_total_allocated_amount(self):
 				base_total_allocated_amount += flt(flt(d.allocated_amount) * flt(d.exchange_rate),
 					self.precision("base_paid_amount"))
 
-		self.total_allocated_amount = abs(total_allocated_amount)
-		self.base_total_allocated_amount = abs(base_total_allocated_amount)
+		if total_allocated_amount > 0:
```

Comment:
I guess this was intentional, only the allocated amount in references should be negative and the total allocated is intentionally kept positive. Just add validation for this
in the `validate_allocated_amount` method. But -300 can still be allocated against -500.
Also please elaborate on what issue allocation like this caused that is still a bit unclear to me